src/explore-education-statistics-admin/src/modules/chart-builder/ChartBuilder.tsx

```diff
@@ -154,6 +147,16 @@ const ChartBuilder = ({
     {},
   );
 
+  const [chartSaveState, setChartSaveState] = React.useReducer(
+    (state, action?: string) => {
+      if (!action) return { saved: false, error: false };
+
+      if (action === 'saved') return { saved: true, error: false };
+      return { saved: true, error: true };
+    },
+    { saved: false, error: false },
+  );
+
```

Comment:
Small change:
The if statements are a little confusing here, and passing any random string or undefined to a reducer seems to break the reducer paradigm. I would expect to pass in one of defined selection of strings as an action

```
(state, action: 'unsaved' | 'saved' | 'error') => {
  switch(action) {
    case 'unsaved':
     /* ... */
  }
}
```

This would help the `setChartSaveState(undefined)` dotted around become more readable as `setChartSaveState('unsaved')`

Reply:
Not sure it's breaking the paradigm, it's a reduce function, undefined is perfectly valid.

Reply:
Changed it and it was a little over the top to used a reducer after that to changed to just using state.... commit incoming...